### PR TITLE
SNOW-1517067 Correct the Index documentation

### PR DESCRIPTION
**Pull request by sfc-gh-vbudati**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1517067

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   The Index documentation incorrectly displayed multiple APIs as partially implemented. They have now been changed to `N`, not implemented. Added tests for the same.

### CODE REVIEW

**Comment by sfc-gh-yzou** [comment]
@sfc-gh-vbudati the test seems failing, please take a look

**Comment by sfc-gh-vbudati** [comment]
@sfc-gh-yzou I think way too many tests are failing from raising NotImplementedError in the Index APIs that called `to_pandas`/were not completely implemented. I think it might be a better idea to just say `N` in the documentation and let the API run with the native pandas warning.
Samedh had a very big PR just to change all of these tests to the lazy Index we have.

**Comment by sfc-gh-helmeleegy** [comment]
> @sfc-gh-yzou I think way too many tests are failing from raising NotImplementedError in the Index APIs that called `to_pandas`/were not completely implemented. I think it might be a better idea to just say `N` in the documentation and let the API run with the native pandas warning. Samedh had a very big PR just to change all of these tests to the lazy Index we have.

I agree. We can treat Index as an exception in this case w.r.t. the diversion between the documentation and actual implementation - until we complete the implementation of the lazy index and bridge this gap.

File: docs/source/modin/supported/index_supported.rst [hunk 1]
```diff
@@ -138,7 +139,7 @@ Methods
 +-----------------------------+---------------------------------+----------------------------------+----------------------------------------------------+
 | ``notna``                   | N                               |                                  |                                                    |
 +-----------------------------+---------------------------------+----------------------------------+----------------------------------------------------+
-| ``astype``                  | P                               |                                  |                                                    |
+| ``astype``                  | P                               |                                  | WIP: SNOW-1458125                                  |
```
Comment (sfc-gh-helmeleegy) [inline docs/source/modin/supported/index_supported.rst:142]: Can we avoid referring to Jira tickets here? I'd say make it `N` for now, and the other PR can turn it into `P`.

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM. Thanks, Varnika!

**Pull request merged**